flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -559,6 +561,57 @@ public void testUpdateUnknownInputChannel() throws Exception {
 		}
 	}
 
+	@Test
+	public void testQueuedBuffers() throws Exception {
+		final NettyShuffleEnvironment network = createNettyShuffleEnvironment();
+
+		final ResultPartition localResultPartition = new ResultPartitionBuilder()
+			.setResultPartitionManager(network.getResultPartitionManager())
+			.setupBufferPoolFactoryFromNettyShuffleEnvironment(network)
+			.build();
+
+		final ResultPartition remoteResultPartition = new ResultPartitionBuilder()
+			.setResultPartitionManager(network.getResultPartitionManager())
+			.setupBufferPoolFactoryFromNettyShuffleEnvironment(network)
+			.build();
+
+		localResultPartition.setup();
+		remoteResultPartition.setup();
+
+		final SingleInputGate inputGate = createInputGate(network, 2, ResultPartitionType.PIPELINED);
+
+		try {
+			final ResultPartitionID localResultPartitionId = localResultPartition.getPartitionId();
+			addUnknownInputChannel(network, inputGate, localResultPartitionId, 0);
+			LocalInputChannel localInputChannel = InputChannelBuilder.newBuilder()
+				.setChannelIndex(0)
+				.setPartitionId(localResultPartitionId)
+				.setupFromNettyShuffleEnvironment(network)
+				.setConnectionManager(new TestingConnectionManager())
+				.buildLocalAndSetToGate(inputGate);
+
+			final ResultPartitionID remoteResultPartitionId = remoteResultPartition.getPartitionId();
+			RemoteInputChannel remoteInputChannel = InputChannelBuilder.newBuilder()
```

Comment:
Move construct `remoteInputChannel` and above `localInputChannel` outside of `try` clause.